### PR TITLE
Add support for -include-checks on changelog

### PR DESCRIPTION
**Pull request by blva**
Changes description
It's important that the changelog correctly captures what users define as breaking changes through `-include-checks`. In order to do that we need to extend `-include-checks` to `-changelog`. To do that I'm defaulting all checkers to log changes as `INFO`, and store overriden log level in the config.

- Extend all optional checkers to default logs to INFO
- If a checker is described in include-checks and it is classified as optional, then that checker will start logging the changes as errors.
- From now on we can always use `allCheckers` and rely that optional checkers will be at `INFO`level unless `include-checks` is used

### CODE REVIEW

File: internal/flags.go [hunk 1]
```diff
@@ -150,7 +150,7 @@ func validateFlags(inputFlags *InputFlags) *ReturnError {
 		return getErrInvalidFlags(fmt.Errorf("\"check-breaking\" and \"changelog\" cannot be used simultaneously"))
 	}
 
-	if len(inputFlags.includeChecks) > 0 && !inputFlags.checkBreaking {
+	if len(inputFlags.includeChecks) > 0 && !(inputFlags.checkBreaking || inputFlags.changelog) {
 		return getErrInvalidFlags(fmt.Errorf("\"include-checks\" is relevant only with \"-check-breaking\""))
```
Comment (ciprian-tibulca) [inline internal/flags.go]:
```suggestion
		return getErrInvalidFlags(fmt.Errorf("\"include-checks\" is relevant only with \"-check-breaking\" or \"-changelog""))
```

File: internal/breaking-changes.go [hunk 2]
```diff
@@ -12,12 +12,12 @@ import (
 func handleBreakingChanges(stdout io.Writer, diffReport *diff.Diff, operationsSources *diff.OperationsSourcesMap, inputFlags *InputFlags) (bool, *ReturnError) {
 	var c checker.BackwardCompatibilityCheckConfig
 	var level checker.Level
+
+	c = checker.GetAllChecks(inputFlags.includeChecks)
+	// Stablish up to what level to log the changes
```
Comment (ciprian-tibulca) [inline internal/breaking-changes.go]:
```suggestion
	// establish up to what level to log the changes
```

File: checker/default_checks.go [hunk 3]
```diff
@@ -10,16 +10,25 @@ func GetDefaultChecks() BackwardCompatibilityCheckConfig {
...
-	return getBackwardCompatibilityCheckConfig(append(defaultChecks(), includedChecks(includeChecks)...))
+	return getBackwardCompatibilityCheckConfig(allChecks(), LevelOverrides(includeChecks))
 }
 
-func GetAllChecks() BackwardCompatibilityCheckConfig {
-	return getBackwardCompatibilityCheckConfig(allChecks())
+func LevelOverrides(includeChecks utils.StringList) map[string]Level {
+	result := map[string]Level{}
+	for _, s := range includeChecks {
+		result[s] = ERR
```
Comment (ciprian-tibulca) [inline checker/default_checks.go:22]:
```suggestion
                // if the checker was explicitly included with the `-include-checks`,
                // it means that it's output is considered a breaking change,
                // so the returned level should overwritten to ERR.
		result[s] = ERR
```

File: internal/breaking-changes.go [hunk 4]
```diff
@@ -12,12 +12,12 @@ import (
 func handleBreakingChanges(stdout io.Writer, diffReport *diff.Diff, operationsSources *diff.OperationsSourcesMap, inputFlags *InputFlags) (bool, *ReturnError) {
 	var c checker.BackwardCompatibilityCheckConfig
 	var level checker.Level
+
+	c = checker.GetAllChecks(inputFlags.includeChecks)
```
Comment (blva) [inline internal/breaking-changes.go:16]: from now on, all checkers will be loaded, but every optional checker will default to `INFO`logs, so they will be filtered out.

**Review by ciprian-tibulca — APPROVED** [review]
LGTM

**Pull request merged**